src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,64 @@
+package duke.parser;
+
+import duke.DukeException;
+import duke.command.*;
+import duke.task.TaskList;
+
+public class Parser {
+
+    private static final String ERROR_INVALID_INPUT = "I'm sorry, but I don't know what that means :-(";
+    private static final String ERROR_MISSING_DESCRIPTION = "The description cannot be empty.";
+    private static final String ERROR_MISSING_TASK_ID = "The id of the task must be provided.";
+    private static final String ERROR_INVALID_TASK_ID = "The id of the task must be a number. e.g. done 1";
+    private static final String ERROR_TOO_MANY_ARGUMENTS = "There are too many arguments for this command.";
+
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] line = fullCommand.split(" ", 2);
+        switch (line[0]) {
```

Comment:
Nice use of switch-case to prevent "arrowhead style code"!! (Not too many nested if-else loops)